### PR TITLE
created a function corresponding to LogPoly in Sims et.al. 2020

### PR DESCRIPTION
**Pull request by nmahesh1412**
Added a logpoly from Peter sims paper. This is so that I can run the cases in table 3 of sims et.al.2020 paper.

### CODE REVIEW

File: src/edges_estimate/foregrounds.py [hunk 1]
```diff
@@ -302,3 +302,44 @@ def calculate(self, ctx, **params):
                     break  # only add to one thing, otherwise it's doubling up.
                 else:
                     ctx[key] *= bias
+
+
+@attr.s
+class LogPoly(Foreground):
```
Comment (nmahesh1412) [inline src/edges_estimate/foregrounds.py:308]: @PSims It would be nice to get your review on this :)

Comment (PSims) [reply src/edges_estimate/foregrounds.py:308]: This looks reasonable to me! 
As an additional check I would suggest building a mock data set constructed to follow this parametrisation and then make sure that you recover the result you expect when fitting it.

**Review by PSims — APPROVED** [review]

File: src/edges_estimate/foregrounds.py [hunk 2]
```diff
@@ -302,3 +302,44 @@ def calculate(self, ctx, **params):
...
+    ----------
+    poly_order : int
+        The maximum polynomial order will be `poly_order `. There are `poly_order`
+        total parameters.
+    """
+
+    poly_order = attr.ib(5, converter=int, kw_only=True)
+
+    def __attrs_post_init__(self):
+        super().__attrs_post_init__()
```
Comment (steven-murray) [inline src/edges_estimate/foregrounds.py:322]: you don't need this

File: src/edges_estimate/foregrounds.py [hunk 3]
```diff
@@ -302,3 +302,44 @@ def calculate(self, ctx, **params):
                     break  # only add to one thing, otherwise it's doubling up.
                 else:
                     ctx[key] *= bias
+
+
+@attr.s
+class LogPoly(Foreground):
+    """
+    LogPoly model from Sims et.al. 2020 (Equation 18)
+    T_{Fg} = 10^sum(d_i*log10(ν/ν_0)^i)_{i=1 to N}
```
Comment (steven-murray) [inline src/edges_estimate/foregrounds.py:311]: why is there no `i=0` term? wouldn't that give the basic power-law behavior?

File: src/edges_estimate/foregrounds.py [hunk 4]
```diff
@@ -302,3 +302,44 @@ def calculate(self, ctx, **params):
...
+    Parameters
+    ----------
+    poly_order : int
+        The maximum polynomial order will be `poly_order `. There are `poly_order`
```
Comment (steven-murray) [inline src/edges_estimate/foregrounds.py:315]: this isn't true given the below implementation... you need to do `range(1, self.poly_order + 1)` to get the maximum order as `self.poly_order`.

**Review by steven-murray — CHANGES_REQUESTED** [review]
One question and one comment to change either the docs or the implementation. I also agree with Peter that you should write a quick test to make sure it fits correctly.

File: src/edges_estimate/foregrounds.py (hunk 3 above)
Comment (nmahesh1412) [reply src/edges_estimate/foregrounds.py:311]: This is a good question. Maybe @PSims can confirm the equation? I followed the equation in the paper.

Comment (PSims) [reply src/edges_estimate/foregrounds.py:311]: That's a good spot, i=0 should be in there (it sets the foreground temperature at the reference frequency). It's listed correctly in Tab. 2.

File: tests/test_foregrounds.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import pytest
+
+import numpy as np
+from yabf import chi2, run_map
+
+
+# @pytest.fixture(scope="session")
```
Comment (steven-murray) [inline tests/test_foregrounds.py:7]: remove the comment here

File: tests/test_foregrounds.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+def test_retrieve_params(fiducial_fg_logpoly):
+    spec = create_mock_data(fiducial_fg_logpoly)
+    lk = chi2.MultiComponentChi2(
+        kind="spectrum", components=[fiducial_fg_logpoly], data=spec
+    )
+    a = run_map(lk)
+    print(a)
+    assert a.success
+    assert len(a.x) == 3
```
Comment (steven-murray) [inline tests/test_foregrounds.py:22]: you should assert something like `assert np.allclose(a.x, [2, -1, 50])` (or whatever you put in)

**Review by steven-murray — APPROVED** [review]
Looks like there's something failing with the deployment. I'll try to sort that out and merge it in, then we can get this in.

**Pull request merged**